commit 5f782dee8d58247f2a277f142a59c227fd0b500e
Author: Jason Cooper <jason@lakedaemon.net>
Date:   Wed Oct 6 10:08:01 2010 -0400

    staging: brcm80211: fix 'do not init statics to 0 or NULL'
    
    Signed-off-by: Jason Cooper <jason@lakedaemon.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

--- a/drivers/staging/brcm80211/brcmfmac/bcmsdh_linux.c
+++ b/drivers/staging/brcm80211/brcmfmac/bcmsdh_linux.c
@@ -70,7 +70,7 @@ struct bcmsdh_hc {
 	spinlock_t irq_lock;
 #endif
 };
-static bcmsdh_hc_t *sdhcinfo = NULL;
+static bcmsdh_hc_t *sdhcinfo;
 
 /* driver info, initialized when bcmsdh_register is called */
 static bcmsdh_driver_t drvinfo = { NULL, NULL };
--- a/drivers/staging/brcm80211/brcmfmac/bcmsdh_sdmmc_linux.c
+++ b/drivers/staging/brcm80211/brcmfmac/bcmsdh_sdmmc_linux.c
@@ -60,7 +60,7 @@ int sdio_function_init(void);
 void sdio_function_cleanup(void);
 
 /* module param defaults */
-static int clockoverride = 0;
+static int clockoverride;
 
 module_param(clockoverride, int, 0644);
 MODULE_PARM_DESC(clockoverride, "SDIO card clock override");
--- a/drivers/staging/brcm80211/brcmfmac/dhd_common.c
+++ b/drivers/staging/brcm80211/brcmfmac/dhd_common.c
@@ -750,7 +750,7 @@ static void wl_show_host_event(wl_event_
 
 	case WLC_E_TRACE:
 		{
-			static uint32 seqnum_prev = 0;
+			static uint32 seqnum_prev;
 			msgtrace_hdr_t hdr;
 			uint32 nblost;
 			char *s, *p;
--- a/drivers/staging/brcm80211/brcmfmac/dhd_linux.c
+++ b/drivers/staging/brcm80211/brcmfmac/dhd_linux.c
@@ -58,8 +58,8 @@ struct semaphore wifi_control_sem;
 
 struct dhd_bus *g_bus;
 
-static struct wifi_platform_data *wifi_control_data = NULL;
-static struct resource *wifi_irqres = NULL;
+static struct wifi_platform_data *wifi_control_data;
+static struct resource *wifi_irqres;
 
 int wifi_get_irq_number(unsigned long *irq_flags_ptr)
 {
@@ -2542,7 +2542,7 @@ int dhd_os_ioctl_resp_wake(dhd_pub_t *pu
 void dhd_os_wd_timer(void *bus, uint wdtick)
 {
 	dhd_pub_t *pub = bus;
-	static uint save_dhd_watchdog_ms = 0;
+	static uint save_dhd_watchdog_ms;
 	dhd_info_t *dhd = (dhd_info_t *) pub->info;
 
 	/* don't start the wd until fw is loaded */
--- a/drivers/staging/brcm80211/brcmfmac/linux_osl.c
+++ b/drivers/staging/brcm80211/brcmfmac/linux_osl.c
@@ -42,7 +42,7 @@ typedef struct bcm_static_buf {
 	unsigned char buf_use[MAX_STATIC_BUF_NUM];
 } bcm_static_buf_t;
 
-static bcm_static_buf_t *bcm_static_buf = 0;
+static bcm_static_buf_t *bcm_static_buf;
 
 #define MAX_STATIC_PKT_NUM 8
 typedef struct bcm_static_pkt {
@@ -51,7 +51,7 @@ typedef struct bcm_static_pkt {
 	struct semaphore osl_pkt_sem;
 	unsigned char pkt_use[MAX_STATIC_PKT_NUM * 2];
 } bcm_static_pkt_t;
-static bcm_static_pkt_t *bcm_static_skb = 0;
+static bcm_static_pkt_t *bcm_static_skb;
 #endif				/* DHD_USE_STATIC_BUF */
 
 typedef struct bcm_mem_link {
--- a/drivers/staging/brcm80211/brcmfmac/wl_cfg80211.c
+++ b/drivers/staging/brcm80211/brcmfmac/wl_cfg80211.c
@@ -47,8 +47,8 @@
 #include <linux/firmware.h>
 #include <wl_cfg80211.h>
 
-static struct sdio_func *cfg80211_sdio_func = NULL;
-static struct wl_dev *wl_cfg80211_dev = NULL;
+static struct sdio_func *cfg80211_sdio_func;
+static struct wl_dev *wl_cfg80211_dev;
 
 uint32 wl_dbg_level = WL_DBG_ERR | WL_DBG_INFO;
 
--- a/drivers/staging/brcm80211/brcmfmac/wl_iw.c
+++ b/drivers/staging/brcm80211/brcmfmac/wl_iw.c
@@ -101,7 +101,7 @@ extern int dhd_wait_pend8021x(struct net
 #define IW_EVENT_IDX(cmd)	((cmd) - IWEVFIRST)
 #endif
 
-static void *g_scan = NULL;
+static void *g_scan;
 static volatile uint g_scan_specified_ssid;
 static wlc_ssid_t g_specific_ssid;
 
